server/webhook/webhook.go

```diff
@@ -165,6 +165,15 @@ func normalizeNamespacedProject(pathWithNamespace string) (namespace string, pro
 	return strings.Join(splits[:len(splits)-1], "/"), splits[len(splits)-1]
 }
 
+// normalizeNamespacedProjectByHomepage converts data from web hooks to format expected by our plugin.
+func normalizeNamespacedProjectByHomepage(homapge string) (namespace string, project string) {
+	splits := strings.Split(homapge, "/")
+	if len(splits) < 2 {
+		return "", ""
+	}
+	return strings.Join(splits[len(splits)-2:4], "/"), splits[len(splits)-1]
```

Comment:
I don't understand the range bounds here: if `len(splits) < 4`, this will panic. Did you mean to omit the `:4` and just take everything left in the slice?

Reply:
actually the condition previously [here](https://github.com/mattermost/mattermost-plugin-gitlab/pull/344/files#diff-4a5a5dac58c1772250e75e67c74e53b8e2eb82b1a7af05edea37154549efe09aR171) needs to change to `if len(splits) < 4 `

Reply:
There's still a panic here: if `len(splits) >= 7`, then the slice amounts to something like `splits[5:4]` which isn't allowed.

Furthermore, if `len(splits) == 6` then the slice amounts to `splits[4:4]`, which would be an empty slice, and an empty string after `strings.Join` -- probably not expected?

Can we unit test this with all the possible inputs?